code/modules/clothing/chameleon.dm

```diff
@@ -420,9 +429,87 @@
 	chameleon_action.chameleon_blacklist = typecacheof(/obj/item/weapon/gun/magic, ignore_root_path = FALSE)
 	chameleon_action.initialize_disguises()
 
+/obj/item/weapon/gun/energy/laser/chameleon/Initialize()
+	..()
+	chameleon_projectile_vars = list("name" = "practice laser", "icon" = 'icons/obj/projectiles.dmi', "icon_state" = "laser")
+	copy_vars = list("name", "icon", "icon_state", "speed", "color", "hitsound", "forcedodge", "impact_effect_type", "range", "suppressed", "hitsound_wall", "impact_effect_type", "pass_flags")
+
 /obj/item/weapon/gun/energy/laser/chameleon/emp_act(severity)
 	chameleon_action.emp_randomise()
 
+/obj/item/weapon/gun/energy/laser/chameleon/proc/set_chameleon_vars(obj/item/projectile/P)
+	chameleon_projectile_vars = list("name" = "practice laser", "icon" = 'icons/obj/projectiles.dmi', "icon_state" = "laser")
+	for(var/V in copy_vars)
+		if(P.vars[V])
+			chameleon_projectile_vars[V] = P.vars[V]
+	if(chambered)
+		for(var/V in chameleon_projectile_vars)
+			if(chambered.vars[V])
+				chambered.vars[V] = chameleon_projectile_vars[V]
+		if(copy_sound)
+			chambered.fire_sound = fire_sound
+
+/obj/item/weapon/gun/energy/laser/chameleon/proc/get_chameleon_projectile(guntype)
+	chameleon_projectile_vars = list()
+	var/obj/item/weapon/gun/G = new guntype(src)
+	if(!istype(G))
+		qdel(G)
+		return FALSE
+	if(copy_burst)
+		burst_size = G.burst_size
+	if(copy_sound)
+		fire_sound = G.fire_sound
+	if(istype(G, /obj/item/weapon/gun/ballistic))
+		var/obj/item/weapon/gun/ballistic/BG = G
+		var/obj/item/ammo_box/AB = new BG.mag_type(src)
+		qdel(BG)
+		if(!istype(AB)||!AB.ammo_type)
+			qdel(AB)
+			return FALSE
+		var/obj/item/ammo_casing/AC = new AB.ammo_type(src)
```

Comment:
When you spawn a `obj/item/ammo_box` it fills itself with ammo so you can access them through `BG` rather than creating them here. https://github.com/tgstation/tgstation/blob/master/code/modules/projectiles/box_magazine.dm#L27
Same goes for the projectile.